packages/hydrooj/src/handler/judge.ts

```diff
@@ -234,17 +269,18 @@ class JudgeConnectionHandler extends ConnectionHandler {
     async newTask() {
         while (!this.closed) {
             /* eslint-disable no-await-in-loop */
-            if (this.processing.length >= this.concurrency) {
-                await sleep(500);
```

Comment:
what about simply change to

```ts
await Promise.race([
  sleep(500),
  new Promise((resolve) => {
    this.callback = resolve;
  });
]);
```

and trigger `this.callback` (if present) on client `end` event?

seems there's no need to add an extra helper class for it.